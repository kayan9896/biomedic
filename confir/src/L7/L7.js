import React, { useState, useEffect, useRef } from 'react';

function L7({handledit, setReport, editable, leftCheckMark, rightCheckMark, recon, setPause}) {
    const imageMap = {
      1: require('./APStatusIcon.png'),
      2: require('./RcnSucceed.png'),
      3: require('./RcnFailed.png'),
      0: require('./OBStatusIcon.png'),
    };
  
    return(
      <div>
        <img src={require('./ImagingModeToolbar.png')} alt="Imaging Mode Toolbar" style={{position:'absolute', top:'458px', left:'921px', zIndex:7}}/>
        
        <img className="image-button"  src={require('./EditIcon.png')} alt="edit icon" style={{position:'absolute', top:'466px', left:'928px', zIndex:7}} onClick={handledit}/> 
        {!editable && <img src={require('./EditIconDis.png')} alt="edit icon dis" style={{position:'absolute', top:'466px', left:'928px', zIndex:7}}/>}
        <img className="image-button"  src={require('./ReportIcon.png')} alt="Report Icon" style={{position:'absolute', top:'547px', left:'928px', zIndex:7}} onClick={()=>{setReport(true)}}/>
        
        
        <>
        <img className="image-button" src={require('./PauseButtonBg.png')} alt="acquire icon" style={{position:'absolute', top:'659px', left:'921px', zIndex:7}}/>
        <img className="image-button"  src={require('./PauseButton.png')} alt="acquire icon" style={{position:'absolute', top:'664px', left:'926px', zIndex:7}} onClick={()=>{setPause(4)}}/>
        </>
        

        {/*Show icon based on backend param*/}
        {rightCheckMark!==null && (
          <img src={imageMap[rightCheckMark]} style={{ position: 'absolute', top: '860px', left: '1020px', zIndex: 7 }} alt="status icon" />
        )}
        {leftCheckMark!==null && (
          <img src={imageMap[leftCheckMark]} style={{ position: 'absolute', top: '860px', left: '815px', zIndex: 7 }} alt="status icon" />
        )}
        {recon!==null && (
          <img src={imageMap[recon]} style={{ position: 'absolute', top: '852px', left: '807px', zIndex: 7 }} alt="recon icon" />
        )}
      </div>
    )
}

export default L7;