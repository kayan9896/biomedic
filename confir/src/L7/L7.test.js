// L7.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // For extended matchers like .toBeInTheDocument()

import L7 from './L7'; // Adjust the path if your component is in a different directory


describe('L7 Component - Static Elements', () => {
  // Section 1: Test Rendering of Static Elements

  it('renders all static toolbar and button background images with correct alt text', () => {
    // Render the component with all props as null or no-op functions for initial test
    render(
      <L7
        handledit={() => {}}
        setReport={() => {}}
        leftCheckMark={null}
        rightCheckMark={null}
        recon={null}
        setPause={() => {}}
      />
    );

    // Verify presence of static images using their alt text
    // Note: ensure the alt texts in the component match these expectations
    expect(screen.getByRole('img', { name: /Imaging Mode Toolbar/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /edit icon/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /Report Icon/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /acquire icon/i,  hidden: true })).toBeInTheDocument(); // For PauseButtonBg
    expect(screen.getByRole('img', { name: /acquire icon/i })).toBeInTheDocument(); // For PauseButton

    // To differentiate the two "acquire icon" elements, you might need to be more specific,
    // or add more descriptive alt text in the component.
    // For now, we're checking if both are present.
    // The `hidden: true` is an attempt to target the background more specifically if needed,
    // but a unique alt text is always better for clarity.
  });

  it('matches snapshot for default rendering (all dynamic icons null)', () => {
    const { asFragment } = render(
      <L7
        handledit={() => {}}
        setReport={() => {}}
        leftCheckMark={null}
        rightCheckMark={null}
        recon={null}
        setPause={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
