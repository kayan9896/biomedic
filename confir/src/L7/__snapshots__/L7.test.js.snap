// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`L7 Component - Static Elements matches snapshot for default rendering (all dynamic icons null) 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="Imaging Mode Toolbar"
      class="image-button"
      src="mock-image-path"
      style="position: absolute; top: 458px; left: 921px; z-index: 7;"
    />
    <img
      alt="edit icon"
      class="image-button"
      src="mock-image-path"
      style="position: absolute; top: 466px; left: 928px; z-index: 7;"
    />
    <img
      alt="Report Icon"
      class="image-button"
      src="mock-image-path"
      style="position: absolute; top: 547px; left: 928px; z-index: 7;"
    />
    <img
      alt="acquire icon"
      class="image-button"
      src="mock-image-path"
      style="position: absolute; top: 659px; left: 921px; z-index: 7;"
    />
    <img
      alt="acquire icon"
      class="image-button"
      src="mock-image-path"
      style="position: absolute; top: 664px; left: 926px; z-index: 7;"
    />
  </div>
</DocumentFragment>
`;
